llvm/lib/Transforms/Scalar/SROA.cpp

```diff
@@ -5178,6 +5188,7 @@ bool SROA::splitAlloca(AllocaInst &AI, AllocaSlices &AS) {
       };
       for_each(findDbgDeclares(Fragment.Alloca), RemoveOne);
       for_each(findDVRDeclares(Fragment.Alloca), RemoveOne);
+      for_each(findDVRValues(Fragment.Alloca), RemoveOne);
```

Comment:
Related to our conversation in the other PR — this is not a place to use findDbgUsers() then?

Reply:
As @SLTozer said in [this thread](https://github.com/llvm/llvm-project/pull/94068#discussion_r1625972446)

We do not want to get assigns, we only want DVRDeclares, DbgDeclares, and now DVRValues, so this usage of `findDVRValues` is correct.